### PR TITLE
✨ Add the concept of a (unique) container ID

### LINKED ISSUE #11: Add the concept of an object-store UUID to identify it
E.g. to associate it with an existing DB, or to uniquely refer to it.

This could be generated during the `init_container` call and then stored in the repository config json

### PR DESCRIPTION
**Pull request by giovannipizzi**
This fixes #11
I also fixed a bug I discovered, where config values that
were cached but the cache was not cleared when re-initialising the
container.
To reduce the risk of such a problem, I am now only caching the *whole*
configuration dictionary (and not each value), and I don't need to
implement caching anymore for each single config value; and I just
need to clear one variable `self._config` rather than each newly-defined
configuration.

In addition, I add a test to check that re-initialising the container
creates a new container ID (`container.container_id`).

### CODE REVIEW

**Comment by giovannipizzi** [comment]
This is a draft PR and is blocked because it builds upon the commits of #96 and can be merged only after #96 is merged

**Review by chrisjsewell — APPROVED** [review]

**Pull request merged**
